yt_dlp/cookies.py

```diff
@@ -0,0 +1,738 @@
+import ctypes
+import json
+import os
+import shutil
+import sqlite3
+import struct
+import subprocess
+import sys
+import warnings
+from datetime import datetime, timedelta
+
+from yt_dlp.aes import aes_cbc_decrypt
+from yt_dlp.compat import compat_cookiejar_Cookie, compat_b64decode, compat_TemporaryDirectory
+from yt_dlp.utils import YoutubeDLCookieJar, expand_path, bytes_to_intlist, intlist_to_bytes
+
+try:
+    from Crypto.Cipher import AES
+    CRYPTO_AVAILABLE = True
+except ImportError:
+    CRYPTO_AVAILABLE = False
+
+try:
+    import keyring
+    KEYRING_AVAILABLE = True
+except ImportError:
+    KEYRING_AVAILABLE = False
+
+
+SUPPORTED_BROWSERS = ['brave', 'chrome', 'chromium', 'edge' 'firefox', 'opera', 'safari', 'vivaldi']
+CHROMIUM_BASED_BROWSERS = {'brave', 'chrome', 'chromium', 'edge' 'opera', 'vivaldi'}
+
+
+class Logger:
+    def debug(self, message):
+        print(message)
+
+    def info(self, message):
+        print(message)
+
+    def warning(self, message):
+        print(message, file=sys.stderr)
+
+    def error(self, message):
+        print(message, file=sys.stderr)
```

Comment:
Not needed. I assume you were using this for debugging?

Reply:
it's an interface, but with trivial default implementations so I implemented each function rather than doing `raise NotImplementedError`

Reply:
ideally it would be nice to use [standard python logging](https://docs.python.org/3/library/logging.html) but I need to adhere to what the rest of the project is doing. I think it would be possible to configure logging so that loggers under `yt_dlp` are handled by the YoutubeDL object but then that assumes that YoutubeDL is a singleton which may not be the case when used as a library